src/libisyntax.h

```diff
@@ -1,3 +1,68 @@
 #pragma once
 
-void libisyntax_init();
+#include <stdint.h>
+
+// API conventions:
+// - return type is one of:
+// -- appropriate data type for simple getters. For complex getters doing work, isyntax_error_t + out variable.
+// -- void if the function doesn't fail at runtime (e.g. destructors).
+// -- isyntax_error_t otherwise.
+// - Output arguments are last, name prefixed with 'out_'.
+// - 'Object operated on' argument is first.
+// - Function names are prefixed with 'libisyntax_', as C doesn't have namespaces. Naming convention
+//   is: libisyntax_<object>_<action>(..), where <object> is omitted for the object representing the isyntax file.
+// - Functions don't check for null/out of bounds, for efficiency. (they may assert, but that is implementation detail).
+// - Booleans are represented as int32_t and prefxied with 'is_' or 'has_'.
```

Comment:
Hmm, why not `bool` from `stdbool.h`?

Reply:
Unfortunately the standard bool does not have a defined size, making it not useful for ABI. More details e.g. here: https://stackoverflow.com/questions/10630080/if-bool-is-a-macro-for-int-why-is-it-a-different-size

Reply:
Ahh, gotcha.